### DISCUSSION TITLE
RocketController - Relays keep going on and off

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16760] [General]
I have a RocketController 4 relay device that is running the latest Tasmota. 
https://rocketcontroller.com/product/controller-esp32r4/

I have it connected nearby so can hear it and noticed that relays keep going on and off without my sending any commands.
I have two questions which I cannot seem to find answers for.

1: What could be causing the interference since this is what seems to be happening and how could I monitor that?

2: Is there a way to subscribe to this device, to get only a message if a relay status has changed, on or off.
By having the ability to monitor only those changes, maybe I could correlate those with something else going on nearby.

**Comment by sfromis** [comment]
Read the console, after setting `Weblog 3` for details, to see what Tasmota is doing.

**Reply by dac2020** [reply]
Ok, I'll do that.

**Reply by dac2020** [reply]
This is what I caught in the overnight log. I left it with all four relays off but came back and now relay 3 and 4 are on.

The device is not communicating with a broker, there is no script doing anything and no timer set.
What ever is happening is either spurious or remote hacking or some kind of problem internally.

My test devices don't have any dashboard password set but I don't see any WebGui connections from any IP before the changes so it's not a hack to my WiFi then messing with the device.

These "RfReceived" are what catch my eye especially because they happen just before the change.

```
06:31:26.534 RFR: Data 0x802 (2050), Bits 14, Protocol 25, Delay 271
06:31:26.541 RSL: RESULT = {"Time":"2022-10-09T06:31:26","RfReceived":{"Data":"0x802","Bits":14,"Protocol":25,"Pulse":271}}
06:31:26.562 RUL: RFRECEIVED#DATA$>02 performs "POWER3 2"
06:31:26.565 SRC: Rule
06:31:26.568 CMD: Grp 0, Cmd 'POWER', Idx 3, Len 1, Pld 2, Data '2'
06:31:26.578 RSL: RESULT = {"POWER3":"ON"}
06:31:26.582 RSL: POWER3 = ON
06:31:27.273 CFG: Saved, Count 119, Bytes 4096

Then power2 is changed status;
07:42:02.947 RFR: Data 0xF04 (3844), Bits 12, Protocol 24, Delay 312
07:42:02.954 RSL: RESULT = {"Time":"2022-10-09T07:42:02","RfReceived":{"Data":"0xF04","Bits":12,"Protocol":24,"Pulse":312}}
07:42:02.973 RUL: RFRECEIVED#DATA$>04 performs "POWER2 2"
07:42:02.975 SRC: Rule
07:42:02.980 CMD: Grp 0, Cmd 'POWER', Idx 2, Len 1, Pld 2, Data '2'
07:42:02.991 RSL: RESULT = {"POWER2":"ON"}
07:42:02.997 RSL: POWER2 = ON
```

**Reply by sfromis** [reply]
Ok you found the explanation, it is receiving RF signals from "somewhere", and you have created rules to do something when getting RF signals. Case closed :grinning: 

Spurious RF is not "hacking", just a horrible outdated standard not useful if reliability is desired.

**Reply by dac2020** [reply]
I appreciate that you know what is happening but I don't since I have no rules or anything else configured on this.
How can I see these rules? Maybe something is left over from some previous configuration.

Reliability is extremely desired. I cannot have outlets going on and off or sprinklers running all night without knowing.
Can this be controlled?

**Reply by sfromis** [reply]
You **have** rules. 
`RUL: RFRECEIVED#DATA$>04 performs "POWER2 2"`

You can display your rules using the command `Rule0`. And if you do not want those rules, you can remove or change them. The current one reacting to anything ending in `04` is extremely imprecise, and is best suited for *wanting* "unexpected" actions.

If those rules were preinstalled by Rocketcontroller, they are just doing a bad job, releasing such crap.

**Reply by dac2020** [reply]
Ok, there are rules but I did not put them there and no one else has access to this device.
I don't even have enough knowledge yet to know how to do it let alone even knowing about this function. First I've heard of this.

```
16:41:40.157 CMD: Rule0
16:41:40.163 CMD: Grp 0, Cmd 'RULE', Idx 0, Len 0, Pld -99, Data ''
16:41:40.169 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":163,"Free":348,"Rules":"ON RfReceived#Data$>08 DO POWER1 2 ENDON ON RfReceived#Data$>04 DO POWER2 2 ENDON ON RfReceived#Data$>02 DO POWER3 2 ENDON ON RfReceived#Data$>01 DO POWER4 2 ENDON"}}
16:41:40.191 RSL: RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
16:41:40.208 RSL: RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
```
I looked at another brand (GoLily I think) that I have, also a 4 relay. Not things I put on the device knowingly at least.
```
17:46:58.280 CMD: Rule0
17:46:58.283 CMD: Grp 0, Cmd 'RULE', Idx 0, Len 0, Pld -99, Data ''
17:46:58.298 MQT: stat/4rNcase/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
17:46:58.319 MQT: stat/4rNcase/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
17:46:58.340 MQT: stat/4rNcase/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
```


Guess I need to look up how to clear these rules then since I definitely don't want things going on and off without knowing.

>The current one reacting to anything ending in 04 is extremely imprecise, and is best suited for wanting "unexpected" actions.

By this, you mean that I could in face have an action of some sort if spurious RF comes in? Does that mean like watching for hacking or simply spurious something on the same network causing interference, etc?

If I could have an action to tell me that there is something like this happening, is there a way to know what the other device is? I assume not without some other monitoring methods.

**Reply by sfromis** [reply]
If you did not create those bad rules, they were preinstalled by some incompetent action of someone from Rocketcontroller.
```
ON RfReceived#Data$>08 DO POWER1 2 ENDON
ON RfReceived#Data$>04 DO POWER2 2 ENDON
ON RfReceived#Data$>02 DO POWER3 2 ENDON
ON RfReceived#Data$>01 DO POWER4 2 ENDON
```
Decent rules would check the full code, but some remotes may not transmit the same code every time for the same button.

What you have is not hacking, but likely someone nearby using a remote transmitting a code happening to end in one of 08,04,02,01. Could in principle also be spurious noise, with a bit of bad luck, but it seems to "too often" for that. Could also be something on the crowded public 433 MHz frequency doing something entirely unrelated, which just "happens" to be similar to a remote control signal. Yes, RF is a horrible outdated standard. In principle, you could set up multiple receivers to triangulate where signals may be coming from, but you'd most probably not get value from doing that.

If you have no remote controls you want to use for the relay unit, just remove the rule set. Or remove all by `Rule0 "`

**Reply by barbudor** [reply]
Rules may have been installed by the manufacturer as part of their factory testing.
Do you know if you have a RF module installed ? I thought they were not available yet.
Whatever, if you do not intend to use RF, you can remove the `RFrecv` from GPIO27 (replacing by None)

**Reply by dac2020** [reply]
Not sure if there is an RF module installed on these. I do control it using an app but not RF buttons.
A second one I bought from Aliexpress was junked so cannot compare against the other. 
I used your suggestion Rule0 " so I'll keep an eye on it.

**Reply by sfromis** [reply]
What you posted clearly shows that the device was picking up something as RF input, which makes it implausible that no RF module was installed.

**Reply by barbudor** [reply]
It's called noise
Not a proof that there is a module
If you look at the other discussion I reported that mine also went crazy with rf received messages while no rf is installed

**Reply by dac2020** [reply]
So mainly spurious then if I'm using the term correctly. Something that the board is picking up causing it to consider interference as commands. 

I've changed it as above and so far, no status change today so seems to be solved.

**Comment by dac2020** [comment]
So I'm back on this topic because something comes to mind and I wonder if someone could provide a little more info.

I understand that some functions are outdated, no longer used and can cause the problems I saw above.

However, it brings up the question of why each outlet might have its own RF function if that's the case.
And, if this is outdated, how was it used in the past?

I ask because I cleared *all* of the rules on all of my Tasmota devices without even thinking about it and wondered if I might have cleared other rules that might have been important to functionality.

**Reply by sfromis** [reply]
RF 433 is outdated because it was never great, and more modern RF technologies are used for current stuff.

Anyway, if you do not want loose antennas on your ESP chip possibly picking up noise, just do not have gpio pins configured unless they are connected to something useful, and you actively want to react to whatever may be happening, including RF.

**Reply by dac2020** [reply]
Ok, so there is nothing to lose by removing all rules on all Tasmota devices then?
And 'just do not have GPIO pins' means configuring the rule right?

**Reply by sfromis** [reply]
There can be much to lose by removing *all* rules, you first have to review what they're doing. And "do not have gpio pins configured" has absolutely nothing to do with rules, it has to do with configure template/module. If you have irrelevant RF pins configured, unconfigure.

**Reply by dac2020** [reply]
I'm using the template so don't have any custom settings.
It was suggested in this thread that I use 'rule0 "' on the relay device and since the others seemed to be getting noise also, I did the same to all of them. So, all of them have the same default rules.

```
19:49:13.349 MQT: stat/84F3EB647217/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
19:49:13.398 MQT: stat/84F3EB647217/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
19:49:13.441 MQT: stat/84F3EB647217/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
```

Rule 1 was the culprit but after using the clearing command, these continue to exist. Does it mean that Tasmota has these by default then?

**Reply by sfromis** [reply]
The whole template **is** custom settings, where you can modify what you want, like unconfiguring noisy pins.

And yes, it is perfectly possible that commands to set initial rules are part of the custom binary from them, reappearing after a full reset. My preference would be to upgrade to standard Tasmota builds, but I don't know if they did other things with their custom binary, which then might be missing.

**Reply by barbudor** [reply]
RocketController doesn't need any specific rule to work
You may need rules for your own usage for local automation but nothing mandatory for RC to work as intended

> Rule 1 was the culprit but after using the clearing command, these continue to exist. Does it mean that Tasmota has these by default then?

Tasmota as 3 Rules set. This is what the command `Rules0` shows you. IT will always show you those 3 lines.
What is key is what is shown for each ruleset:
- The "State" telling if the ruleset is active
- The "Rules" which list the rules inside the ruleset
- The "Once" parameter as well as some other details about memory used
In your case, all 3 ruleset are empty and OFF

Here is a exemple of what Rule0 display on one of my plug with energy metring. Rules are intended to power off the plug once the power consumption fall below a given threshold:
```
22:42:48.974 CMD: rule0
22:42:48.982 MQT: stat/skt-barbuxps/RULE = {"Rule1":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":94,"Free":417,"Rules":"on Energy#Power<%Mem1% do ruletimer1 %Mem3% endon on Energy#Power>%Mem2% do ruletimer1 0 endon"}}
22:42:49.000 MQT: stat/skt-barbuxps/RULE = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":35,"Free":476,"Rules":"on rules#timer=1 do power off endon"}}
22:42:49.016 MQT: stat/skt-barbuxps/RULE = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
```
You can read the rules that are in ruleset 1 and ruleset 2 which both are ON. Ruleset3 is empty and OFF
